sorting/heap_sort.cpp

```diff
@@ -30,13 +30,16 @@
  * sorting.
  *
  * @param arr array to be printed
- * @param sz size of array
+ * @param size size of array
  *
  */
 template <typename T>
-void printArray(T *arr, int sz) {
-    for (int i = 0; i < sz; i++) std::cout << arr[i] << "  ";
-    std::cout << "\n";
+void printArray(T *arr, unsigned size) {
```

Comment:
as long as you are doing the refactoring, it might be better to do it correctly. the correct data type for size of an array and position in an array is not `unsigned` but rather `uint32_t`.